src/course/thread-net/sensor-app.md

```diff
@@ -0,0 +1,218 @@
+
+You can connect to your board's serial console using `tockloader` or
+any other serial console application. If your development board
+presents two console devices, the lower-numbered one is usually
+correct. Select 115200 baud, 1 stop bit, no partity, no flow
+control. The following command should also do the trick:
+
+```
+tockloader listen
+```
+
+By default, a Tock board without any applications will respond with a
+message similar to:
+
+```
+Initialization complete. Entering main loop
+NRF52 HW INFO: Variant: AAC0, Part: N52840, Package: QI, Ram: K256, Flash: K1024
+tock$
+```
+
+If you don't see this prompt, try hitting ENTER or pressing the
+`RESET` button on your board (near the left-hand side USB port). In
+case you see the following selection dialog, the nRF52840DK exposes
+the chip's serial console on the first UART port (e.g., `ttyACM0`
+instead of `ttyACM1`). If that does not work, simply try the available
+ports:
+
+```
+> tockloader listen
+[INFO   ] No device name specified. Using default name "tock".
+[INFO   ] No serial port with device name "tock" found.
+[INFO   ] Found 2 serial ports.
+Multiple serial port options found. Which would you like to use?
+[0]     /dev/ttyACM1 - J-Link - CDC
+[1]     /dev/ttyACM0 - J-Link - CDC
+
+Which option? [0] 1
+[INFO   ] Using "/dev/ttyACM0 - J-Link - CDC".
+Initialization complete. Entering main loop
+NRF52 HW INFO: Variant: AAC0, Part: N52840, Package: QI, Ram: K256, Flash: K1024
+tock$
+
+```
+
+The small shell above is called the "process console". It allows you
+to start and stop applications, and control other parts of the tock
+kernel. For instance, `reset` will completely reset your chip,
+re-printing the above greeting. Use `help` to obtain a list of
+commands.
+
+> **Checkpoint:** You can interact with the process console.
+
+## Compiling and Installing an Application
+
+With the kernel running we can now load applications onto our
+board. Tock applications are compiled and loaded separately from the
+kernel. For this tutorial we will use the `libtock-c` userspace
+library, whose source is located outside of the kernel repository
+[here](https://github.com/tock/libtock-c).
+
+We provide some scaffolding for this tutorial. Make sure to enter the
+following directory:
+
+```
+$ cd libtock-c/examples/tutorials/thread_network
+$ ls
+00_sensor_hello
+01_sensor_ipc
+[...] TODO update directory listing!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+```
+
+These applications represent checkpoints for different milestones of
+this tutorial. If you are ever stuck on something, you can try running
+or looking at the subsequent checkpoint. We'll start the tutorial off
+at checkpoint `00_sensor_hello`. Whenever we reach a checkpoint, we
+indicate this through a message like the following:
+
+> **CHECKPOINT:** `00_sensor_hello`
+
+To compile and flash this application, we enter into its directory and
+run the following command:
+
+```
+$ cd 00_sensor_hello
+$ make -j4 install
```

Comment:
Did we explain what the ` -j4` flag does?

Reply:
We can leave it out, it's just to speed up the compilation process by using 4 cores. It's a standard Make flag -- if it's confusing, I'd rather leave it out than explain it.